types/types.hh

```diff
@@ -986,6 +989,12 @@ data_value::data_value(std::optional<NativeType> v)
         : data_value(v ? data_value(*v) : data_value::make_null(data_type_for<NativeType>())) {
 }
 
+//in this case we want to turn it automatically to bool
+template <typename Tag> 
+data_value::data_value(bool_class<Tag> v) : data_value(bool(v)) {
```

Comment:
should probably include something similar to
`template <> inline thread_local const data_type& data_type_for_v<utils::UUID> = uuid_type;`
so that this
`auto ts = value_cast<utils::UUID>(data_type_for<utils::UUID>()->deserialize(*row[ts_index]));` could work for `bool_class`

Reply:
Then we would be able to write `row.get_as<raft::can_vote_t>("can_vote")}` instead of `raft::can_vote_t(row.get_as<bool>("can_vote"))}`

Reply:
yes, seems this machinery doesn't work with templates